dcos/cosmos.py

```diff
@@ -163,33 +159,20 @@ def _cosmos_request(self,
         :return: response returned by calling cosmos at url
         :rtype: requests.Response
         """
-        try:
-            headers = headers_preference[0]
-            if http_request_type is 'post':
-                response = http.post(
-                    url, data=data, json=json, headers=headers, **kwargs)
-            else:
-                response = http.get(
-                    url, data=data, json=json, headers=headers, **kwargs)
-            if not _matches_expected_response_header(headers,
-                                                     response.headers):
-                raise DCOSException(
-                    'Server returned incorrect response type, '
-                    'expected {} but got {}'.format(
-                        headers.get('Accept'),
-                        response.headers.get('Content-Type')))
-            return response
-        except DCOSBadRequest as e:
-            if len(headers_preference) > 1:
-                # reattempt with one less item in headers_preference
-                return self._cosmos_request(url,
-                                            http_request_type,
-                                            headers_preference[1:],
-                                            data,
-                                            json,
-                                            **kwargs)
-            else:
-                raise e
+        if http_request_type is 'post':
+            response = http.post(
+                url, data=data, json=json, headers=headers, **kwargs)
+        else:
+            response = http.get(
+                url, data=data, json=json, headers=headers, **kwargs)
```

Comment:
HTTP GET request are not allowed to have a body. You can remove `data=data` and `json=json`.